### PR TITLE
Package/microservice essentials

### PR DESCRIPTION
**Pull request by seboste**
Specify library name and version:  **microservice-essentials/0.1.0**

I am the author of this C++ open source library that takes care of typical recurring concerns that occur in microservice development.
I need this library to be retrievable from conan center because my open source CLI tool sapper (https://github.com/seboste/sapper) that enables you to rapidly create, extend, and update C++ microservices heavily relies on conan. 

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/microservice-essentials/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        cmake = CMake(self)
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["microservice-essentials"]
+        if self.settings.os != "Windows":
+            self.cpp_info.system_libs = ["pthread"]
+        lib_path = os.path.join(self.package_folder, "lib")
+        self.output.info("Appending PATH environment variable with : {0}".format(lib_path))
+        self.runenv_info.prepend_path("PATH", lib_path)
```
Comment (czoido) [inline recipes/microservice-essentials/all/conanfile.py:106]: Why is this needed?

Comment (seboste) [reply recipes/microservice-essentials/all/conanfile.py:106]: Windows/Shared lib case: .dll is in the lib_path. The test_package (and any other user of the library) needs to be able to find the .dll file. Is there a better way to ensure this?

Comment (czoido) [reply recipes/microservice-essentials/all/conanfile.py:106]: Ok, I see. This is not needed because Conan will add those paths to the `VirtualRunEnv` generator and create a `conanrun.bat/sh` file so you can activate the environment before running the program built in shared mode. Please check this section of the tutorial in the Conan docs: https://docs.conan.io/2/tutorial/consuming_packages/different_configurations.html#modifying-options-linking-the-application-dependencies-as-shared-libraries

File: recipes/microservice-essentials/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    def package(self):
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["microservice-essentials"]
+        if self.settings.os != "Windows":
+            self.cpp_info.system_libs = ["pthread"]
```
[the remark this answers is not in the document]
Comment (seboste) [reply recipes/microservice-essentials/all/conanfile.py]: thanks. Done.

File: recipes/microservice-essentials/all/conanfile.py (hunk 1 above)
Comment (seboste) [reply recipes/microservice-essentials/all/conanfile.py]: unfortunately, just removing isn't enough. The dll needs to be in the bin folder for this to work. I now explicitly copied the files into this folder.

File: recipes/microservice-essentials/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        }
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.16.3 <4]")
+
+    def requirements(self):
+        if self.options.with_examples:
+            self.requires("cpp-httplib/0.12.4")
+            self.requires("nlohmann_json/3.11.2")
+            self.requires("openssl/3.1.0")
```
[the remark this answers is not in the document]
Comment (seboste) [reply recipes/microservice-essentials/all/conanfile.py]: done.

File: recipes/microservice-essentials/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def package(self):
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        copy(self, "*.dll", src=self.source_folder, dst=os.path.join(self.package_folder, "bin"), keep_path=False)
```
Comment (czoido) [inline recipes/microservice-essentials/all/conanfile.py:99]: I think you need this because probably the CMakeLists is installing the `dll's` to other place that's not `bin` (maybe also to `lib`?) which is the default, so in the end the dll's will end up in two places, the ones copied by `cmake.install()` and the ones by the `copy()`. You an set the `cpp_info.bindirs = ["lib"]` if that's the case but the expected thing would be that the dll's end up in the `bin` folder

Comment (seboste) [reply recipes/microservice-essentials/all/conanfile.py]: CMake installs the dlls to the lib folder. That can be changed some time in the future but not for that initial 0.1.0 version. Other recipes use CMakeLists.txt patches for the same issue. If possible, I'd like to avoid that. Thus, I updated the .bindirs property. Maybe not perfect, but better than deploying two copies of the dlls.

**Review by czoido — DISMISSED** [review]

File: recipes/microservice-essentials/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        "with_examples": False
+    }
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+            "Visual Studio": "15.7",
```
Comment (RubenRBS) [inline recipes/microservice-essentials/all/conanfile.py:38]: Missing `msvc` compiler version check :)

```suggestion
            "Visual Studio": "15.7",
            "msvc": "191",
```

**Review by RubenRBS — CHANGES_REQUESTED** [review]

**Review by czoido — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]
